conan/internal/deploy.py

```diff
@@ -80,6 +81,72 @@ def full_deploy(graph, output_folder):
         _deploy_single(dep, conanfile, output_folder, folder_name)
 
 
+def runtime_deploy(graph, output_folder):
+    """
+    Deploy all the shared libraries and the executables of the dependencies in a flat directory.
+    """
+    conanfile = graph.root.conanfile
+    conanfile.output.info(f"Deploying the runtime...")
+    for _, dep in conanfile.dependencies.items():
+        conanfile.output.verbose(f"Searching for shared libraries and executables in {dep.ref}...")
+        if dep.package_folder is None:
+            conanfile.output.verbose(f"{dep.ref} does not have any package folder")
+            continue
+        if not dep.cpp_info.bindirs and not dep.cpp_info.libdirs:
+            conanfile.output.verbose(f"{dep.ref} does not have any bin or lib directory")
+            continue
+
+        for bindir_name in dep.cpp_info.bindirs:
+            bindir_path = os.path.join(dep.package_folder, bindir_name)
+            if not os.path.isdir(bindir_path):
+                conanfile.output.verbose(f"{bindir_path} does not exist")
+                continue
+            file_count = _flatten_directory(dep, conanfile, bindir_path, output_folder)
+            conanfile.output.info(f"Copied {file_count} files from {dep.ref}, directory named {bindir_name}")
+
+        for libdir_name in dep.cpp_info.libdirs:
+            libdir_path = os.path.join(dep.package_folder, libdir_name)
+            if not os.path.isdir(libdir_path):
+                conanfile.output.verbose(f"{libdir_path} does not exist")
+                continue
+            file_count = _flatten_directory(dep, conanfile, libdir_path, output_folder, [".dll", ".dylib",".so"])
```

Comment:
yes maybe

Reply:
I'am working on the onnxruntime recipe and I noticed that they create dll in the lib directory.
From what I understand they create some "modules" that are supposed to loaded at runtime, and they are considered as "LIBRARY" and not "RUNTIME" by cmake during the installation: https://cmake.org/cmake/help/book/mastering-cmake/chapter/Install.html#installing-targets

Note that according to their documentation the main dll shall be next the to the modules: https://onnxruntime.ai/docs/build/eps.html#loading-the-shared-providers so for this specific recipe we shall move the dll in the bin dir.
What is the correct solution here?
- Patch the onnxruntime cmake file so that they are copied to the bin folder?
- In the conanfile:package method, after the cmake install move the dll from the lib to the bin folder?

I am asking in this thread because it shows that a normal cmake install with modules create dll in the lib folder so maybe we have to manage this case in the runtime deployer.